internals/secrethub/audit.go

```diff
@@ -82,30 +129,170 @@ func (cmd *AuditCommand) run() error {
 			return err
 		}
 
-		fmt.Fprint(tabWriter, strings.Join(row, "\t")+"\n")
+		err = formatter.WriteRow(row)
+		if err == errPagerClosed {
+			break
+		} else if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type tableFormatter interface {
+	WriteRow([]string) error
+}
 
-		if i == cmd.perPage {
-			err = tabWriter.Flush()
-			if err != nil {
-				return err
-			}
-			i = 0
+// newJSONFormatter returns a table formatter that formats the given table rows as json.
+func newJSONFormatter(writer io.Writer, fieldNames []string) *jsonFormatter {
+	return &jsonFormatter{encoder: json.NewEncoder(writer), fields: fieldNames}
+}
 
-			// wait for <ENTER> to continue.
-			_, err := ui.Ask(cmd.io, "Press <ENTER> to show more results. Press <CTRL+C> to exit.")
-			if err != nil {
-				return err
-			}
-			fmt.Fprint(tabWriter, header)
+type jsonFormatter struct {
+	encoder *json.Encoder
+	fields  []string
+}
+
+// formatRow returns the json representation of the given row
+// with the configured field names as keys and the provided values
+func (f *jsonFormatter) WriteRow(row []string) error {
+	if len(f.fields) != len(row) {
+		return fmt.Errorf("unexpected number of json fields")
+	}
+
+	jsonMap := make(map[string]string)
+	for i, element := range row {
+		jsonMap[f.fields[i]] = element
+	}
+
+	return f.encoder.Encode(jsonMap)
+}
+
+// newColumnFormatter returns a table formatter that aligns the columns of the table.
+func newColumnFormatter(writer io.Writer, tableWidth int, columns []tableColumn) *columnFormatter {
+	return &columnFormatter{writer: writer, tableWidth: tableWidth, columns: columns}
+}
+
+type columnFormatter struct {
+	tableWidth           int
+	writer               io.Writer
+	computedColumnWidths []int
+	columns              []tableColumn
+	didPrintHeader       bool
```

Comment:
Nitpick: if you want to leave out `did`, `printedHeader` or `headerPrinted` works just as well. (feel free to ignore)

Reply:
Yeah, that sounds better. I'll use `headerPrinted`.